### PR TITLE
Handle regions where metrics are unsupported

### PR DESCRIPTION
**Pull request by djberg96**
It seems that even if a region has the Microsoft.Insights provider enabled, it still does not necessarily support collection of certain metrics in certain regions, e.g. Germany.

In practice what we see is a `Azure::Armrest::BadRequestException` raised in such cases, combined with a misleading error message from Azure. The error message suggests that it's simply a matter of adjusting the `api-version` string, but in reality none of them work if the region simply isn't supported.

Instead of raising an error, this PR adjusts the `MetricsCapture#get_counters` method to explicitly check for a `Azure::Armrest::BadRequestException`, logs a warning, and returns an empty array.

Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1645117

### CODE REVIEW

**Comment by JPrause** [comment]
@miq-bot add_label blocker

**Comment by JPrause** [comment]
@bronaghs as this is for a blocker, can you review asap.

File: app/models/manageiq/providers/azure/cloud_manager/metrics_capture.rb [hunk 1]
```diff
@@ -209,6 +209,11 @@ def get_counters(metrics_conn)
           .list('Microsoft.Compute', 'virtualMachines', target.name, target.resource_group.name)
           .select { |m| m.name.value.in?(COUNTER_NAMES) }
       end
+    rescue ::Azure::Armrest::BadRequestException # Probably means region is not supported
```
Comment (agrare) [inline app/models/manageiq/providers/azure/cloud_manager/metrics_capture.rb]: Is there anything more specific on why it failed in the message that we could check?  something like `raise unless err.message =~ /Region does not support metrics/` or whatever the exception message says.

Comment (agrare) [reply app/models/manageiq/providers/azure/cloud_manager/metrics_capture.rb]: > The error message suggests that it's simply a matter of adjusting the api-version string, but in reality none of them work if the region simply isn't supported.

Oh hm okay

Comment (djberg96) [reply app/models/manageiq/providers/azure/cloud_manager/metrics_capture.rb]: @agrare yeah, unfortunately the error message is not friendly.

**Comment by djberg96** [comment]
@agrare More difficult than you would think, since it may just be a matter of the Microsoft.Insights service not being enabled (which we already check for). What could happen is that the Insights service isn't enabled, the region would be tagged as "unsupported", then later they enable Insights and wonder why metrics aren't being collected.

**Comment by simaishi** [comment]
Hammer backport details:

```
$ git log -1
commit 3e5ba61abfd42d3a3b13d135b2aa2a04c98b4162
Author: Adam Grare <agrare@redhat.com>
Date:   Thu Nov 8 10:55:32 2018 -0500

    Merge pull request #302 from djberg96/metrics_unsupported
    
    Handle regions where metrics are unsupported
    
    (cherry picked from commit a94f2b03a21ac70f07e11ef327740e6af8d889ce)
    
    Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1645117
```

**Comment by simaishi** [comment]
Gaprindashvili backport details:

```
$ git log -1
commit 65952ca7b547d1f9bb9f74fc4d52124ab700963b
Author: Adam Grare <agrare@redhat.com>
Date:   Thu Nov 8 10:55:32 2018 -0500

    Merge pull request #302 from djberg96/metrics_unsupported
    
    Handle regions where metrics are unsupported
    
    (cherry picked from commit a94f2b03a21ac70f07e11ef327740e6af8d889ce)
    
    Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1648955
```

**Pull request merged**